common/v2/features/Dashboard/components/RecentTransactionList.tsx

```diff
@@ -1,79 +1,206 @@
-import React from 'react';
-import { Link } from 'react-router-dom';
-import { Address, CollapsibleTable } from '@mycrypto/ui';
+import React, { useContext } from 'react';
+import { Address } from '@mycrypto/ui';
+import styled from 'styled-components';
 
-import { Amount, DashboardPanel } from 'v2/components';
+import { Amount, DashboardPanel, NewTabLink, AssetIcon, CollapsibleTable } from 'v2/components';
+import { truncate, convertToFiat } from 'v2/utils';
+import { ITxReceipt, TTicker, ITxStatus, StoreAccount, Asset, TSymbol } from 'v2/types';
+import { RatesContext, AddressBookContext, getLabelByAddressAndNetwork } from 'v2/services';
+import { translateRaw } from 'v2/translations';
+import {
+  getTxsFromAccount,
+  txIsFailed,
+  txIsPending,
+  txIsSuccessful
+} from 'v2/services/Store/helpers';
+
+import NoTransactions from './NoTransactions';
 import TransactionLabel from './TransactionLabel';
 import './RecentTransactionList.scss';
-
 import newWindowIcon from 'common/assets/images/icn-new-window.svg';
-import { truncate } from 'v2/utils';
-import { ExtendedAccount, AddressBook } from 'v2/types';
-import { translateRaw } from 'translations';
+import transfer from 'common/assets/images/transactions/transfer.svg';
+import inbound from 'common/assets/images/transactions/inbound.svg';
+import outbound from 'common/assets/images/transactions/outbound.svg';
 
 interface Props {
   className?: string;
-  accountsList: ExtendedAccount[];
-  readAddressBook(uuid: string): AddressBook;
+  accountsList: StoreAccount[];
+}
+
+enum ITxType {
+  TRANSFER = 'TRANSFER',
+  OUTBOUND = 'OUTBOUND',
+  INBOUND = 'INBOUND'
 }
 
-export default function RecentTransactionList({
-  accountsList,
-  readAddressBook,
-  className = ''
-}: Props) {
+export const deriveTxType = (accountsList: StoreAccount[], tx: ITxReceipt) => {
+  const fromAccount = accountsList.find(
+    account => account.address.toLowerCase() === tx.from.toLowerCase()
+  );
+  const toAccount = accountsList.find(
+    account => account.address.toLowerCase() === tx.to.toLowerCase()
+  );
+  return !fromAccount || !toAccount
+    ? fromAccount
+      ? ITxType.OUTBOUND
+      : ITxType.INBOUND
+    : ITxType.TRANSFER;
+};
+
+export const getTxIcon = (type: ITxType) => {
+  switch (type) {
+    case ITxType.TRANSFER:
+      return transfer;
+    case ITxType.INBOUND:
+      return inbound;
+    case ITxType.OUTBOUND:
+      return outbound;
+  }
+};
```

Comment:
In a effort to simplify the code, how about we create a single config object for TxType:
```
{
   [ITxType.TRANSFER]: {
      color: ...,
      icon:  require(...),
      text: <Translate const>  
   }
}
```
So we can reduce the amount of switches by simply calling `TxType[ITxType.TRANSFER].icon`